Remotion/ObjectBinding/Web.UnitTests/UI/Controls/Validation/BocListValidationResultDispatchingValidatorTest.cs

```diff
@@ -160,6 +160,170 @@ public void DispatchValidationFailures_WithRowEditModeControlHavingInvalidValida
+
+      var validator = new BocListValidationResultDispatchingValidator();
+      validator.ControlToValidate = "someBocList";
+      bocList.ID = "someBocList";
+      bocList.SwitchListIntoEditMode();
+
+      Assert.That(bocList.IsListEditModeActive, Is.True);
+
+      var editableRow = ((IBocList)bocList).EditModeController.GetEditableRow(0);
+      ((EditableRow)editableRow).EnsureValidatorsRestored();
+
+      Assert.That(editableRow.HasValidators(), Is.True);
+
+      Assert.That(editableRow.HasValidators(0), Is.True);
+      ControlCollection editableRowValidators = editableRow.GetValidators(0);
+      Assert.That(
+          editableRowValidators.Cast<BaseValidator>().Select(v => v.GetType()),
+          Is.EqualTo(
+              new[]
+              {
+                typeof(ControlCharactersCharactersValidator),
+                typeof(BusinessObjectBoundEditableWebControlValidationResultDispatchingValidator)
+              }));
+
+      ((BaseValidator)editableRowValidators[0]).Validate();
+
+      NamingContainer.Controls.Add(bocList);
+      NamingContainer.Controls.Add(validator);
+
+      var validationResultMock = new Mock<IBusinessObjectValidationResult>();
+      validationResultMock
+          .Setup(e => e.GetValidationFailures(bocList.DataSource.BusinessObject, bocList.Property, true))
+          .Returns(Array.Empty<BusinessObjectValidationFailure>())
+          .Verifiable();
+
+      validationResultMock
+          .Setup(_ => _.GetValidationFailures((IBusinessObject)values[0], simpleColumnDefinition.GetPropertyPath().Properties.Single(), true))
+          .Returns(Array.Empty<BusinessObjectValidationFailure>())
+          .Verifiable();
+      validationResultMock
+          .Setup(_ => _.GetUnhandledValidationFailures(It.IsAny<IBusinessObject>(), It.IsAny<bool>(), It.IsAny<bool>()))
+          .Returns(Array.Empty<BusinessObjectValidationFailure>());
+
+      validator.DispatchValidationFailures(validationResultMock.Object);
+
+      var bocListValidationFailureRepository = bocList.ValidationFailureRepository;
+
+      var failuresInRepository = bocListValidationFailureRepository.GetUnhandledValidationFailuresForBocListAndContainingDataRowsAndDataCells(true);
+      Assert.That(failuresInRepository.Count, Is.EqualTo(1));
+      Assert.That(failuresInRepository.Single().ColumnDefinition, Is.EqualTo(simpleColumnDefinition));
+      Assert.That(failuresInRepository.Single().RowObject, Is.EqualTo(values[0]));
+      Assert.That(failuresInRepository.Single().Failure.ErrorMessage, Is.EqualTo("The value contains unsupported characters: \"\u0001\" (position: 1)."));
+
+      validationResultMock.Verify();
+
+      // Validates that dispatch was called to dispatching validator
+      var bocTextValue = ((Control)editableRow).FindControl("0") as BusinessObjectBoundEditableWebControl;
+      validationResultMock.Verify(_ => _.GetValidationFailures(bocTextValue.DataSource.BusinessObject, bocTextValue.Property, true));
```

Comment:
Does Moq implictly reset the recordings after a verify? Otherwise, we test the same thing we do in line 231 with the validationResultMock.Verify call, correct?

Reply:
These calles were already handled in the above Setup that uses `simpleColumnDefinition.GetPRopertyPath()...`, this was therefore an unrequired verify.